src/core/server/csp/csp_directives.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CspConfigType } from './config';
+
+export type CspDirectiveName = 'script-src' | 'worker-src' | 'style-src' | 'frame-ancestors';
+
+export const defaultRules: Partial<Record<CspDirectiveName, string[]>> = {
+  'script-src': [`'unsafe-eval'`, `'self'`],
+  'worker-src': [`blob:`, `'self'`],
+  'style-src': [`'unsafe-inline'`, `'self'`],
+};
+
+export class CspDirectives {
+  private readonly directives = new Map<CspDirectiveName, Set<string>>();
+
+  addDirectiveValue(directiveName: CspDirectiveName, directiveValue: string) {
+    if (!this.directives.has(directiveName)) {
+      this.directives.set(directiveName, new Set());
+    }
+    this.directives.get(directiveName)!.add(directiveValue);
```

Comment:
Using a `Set` here to 'silently' remove potential duplicates. Felt like it made more sense than throwing in that case.